From 7fbf9968ec5ed62e60872b28e495e26c797f1e50 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Mon, 11 Oct 2021 17:07:20 +0800
Subject: [PATCH] fix grub efi boot failure

- Tweak efi_dir according to oe-core's EFI layout

- Drop grub-mkrelpath which is not necessary for oe-core's EFI layout

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/modules/storage/bootloader/efi.py | 15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/pyanaconda/modules/storage/bootloader/efi.py b/pyanaconda/modules/storage/bootloader/efi.py
index 07c264ba9..c7e33bbd0 100644
--- a/pyanaconda/modules/storage/bootloader/efi.py
+++ b/pyanaconda/modules/storage/bootloader/efi.py
@@ -42,6 +42,16 @@ class EFIBase(object):
     def _efi_config_dir(self):
         return "efi/EFI/BOOT"
 
+    @property
+    def _config_dir(self):
+        """ Full path to configuration directory. """
+        return self._efi_config_dir
+
+    @property
+    def config_file(self):
+        """ Full path to configuration file. """
+        return "%s/grub/%s" % (self.config_dir, self._config_file)
+
     def efibootmgr(self, *args, **kwargs):
         if not conf.target.is_hardware:
             log.info("Skipping efibootmgr for image/directory install.")
@@ -211,11 +221,6 @@ class EFIGRUB(EFIBase, GRUB2):
             if fs_uuid is None:
                 raise BootLoaderError("Could not get stage2 filesystem UUID")
 
-            grub_dir = util.execWithCapture("grub-mkrelpath", [grub_dir],
-                                            root=conf.target.system_root)
-            if not grub_dir:
-                raise BootLoaderError("Could not get GRUB directory path")
-
             fd.write("search --no-floppy --fs-uuid --set=dev %s\n" % fs_uuid)
             fd.write("set prefix=($dev)%s\n" % grub_dir)
             fd.write("export $prefix\n")
-- 
2.27.0

